src/index.ts

```diff
@@ -37,187 +17,48 @@ enum ErrorCode {
-				archive,
-				additionalJsons,
-				descriptor.runners
-			);
+	const config = await startup();
 
-			// TODO: We can ask for environment variables here too and cache them
-			/*
-			const { enableEnv } = await prompt<{ enableEnv: boolean }>([
-				{
-					type: 'confirm',
-					name: 'enableEnv',
-					message: 'Add env vars?',
-					default: false
-				}
-			]);
-			const env = enableEnv
-				? await prompt<{ env: string }>([
-						{
-							type: 'input',
-							name: 'env',
-							message: 'Type env vars in the format: K1=V1, K2=V2'
-						}
-				]).then(({ env }) =>
-						env
-							.split(',')
-							.map(kv => {
-								const [k, v] = kv.trim().split('=');
-								return { [k]: v };
-							})
-							.reduce((obj, kv) => Object.assign(obj, kv), {})
-				)
-				: {};
-			*/
-
-			info(`Deploying ${rootPath}...\n`);
-
-			try {
-				await api.deploy(name, [], plan);
-			} catch (err) {
-				apiError(err as AxiosError);
-			}
-
-			// TODO: Need a TUI for logs
-
-			try {
-				const container: string = await containerSelection([
-					...descriptor.runners,
-					'deploy'
-				]);
-				const type =
-					container === 'deploy' ? LogType.Deploy : LogType.Job;
-
-				await logs(container, name, type);
-			} catch (err) {
-				apiError(err as AxiosError);
-			}
-		};
-
-		const createJsonAndDeploy = async (saveConsent: string) => {
-			const potentialPackages = generateJsonsFromFiles(descriptor.files);
-			const potentialLanguages = Array.from(
-				new Set<LanguageId>(
-					potentialPackages.reduce<LanguageId[]>(
-						(langs, pkg) => [...langs, pkg.language_id],
-						[]
-					)
-				)
-			);
-
-			const languages = await languageSelection(potentialLanguages);
-			const packages = potentialPackages.filter(pkg =>
-				languages.includes(pkg.language_id)
-			);
-
-			for (const pkg of packages) {
-				pkg.scripts = await fileSelection(
-					`Select files to load with ${printLanguage(
-						pkg.language_id
-					)}`,
-					pkg.scripts
-				);
-			}
-
-			const cacheJsons = saveConsent === 'Y' || saveConsent === 'YES';
-
-			const additionalPackages = cacheJsons
-				? await (async () => {
-						for (const pkg of packages) {
-							await fs.writeFile(
-								join(
-									rootPath,
-									`metacall-${pkg.language_id}.json`
-								),
-								JSON.stringify(pkg, null, 2)
-							);
-						}
-
-						// If they are cached, genearte the descriptor again
-						descriptor = await generatePackage(rootPath);
-
-						// The descriptor already contains the packages so
-						// there is no need to send additional packages
-						return [];
-				  })()
-				: packages; // Otherwise, packages are not cached, send them
+	//if addrepo is passed then deploy from repository url
+	if (args['addrepo']) {
+		try {
+			await deployFromRepository(config, plan);
+		} catch (e) {
+			console.error(e);
+		}
+	}
 
-			await deploy(additionalPackages);
-		};
+	//if workdir is passed call than deploy using package
+	if (args['workdir']) {
```

Comment:
```suggestion
	if (args['workdir']) {
```
Isn't this in your latest commit? workdir have a default value, will be true always